custom-stylelint-plugins/spectrum-class-name-rule.js

```diff
@@ -0,0 +1,37 @@
+const stylelint = require('stylelint');
+
+const ruleName = 'custom-rule/spectrum-class-name';
+const messages = stylelint.utils.ruleMessages(ruleName, {
```

Comment:
A new plugin should not be needed for this - we can pass the customizations in to https://stylelint.io/user-guide/rules/selector-class-pattern/:

```
"selector-class-pattern": [
  "^spectrum(-[A-Z][a-zA-Z0-9]+)", {
    resolveNestedSelectors: true
  }
],
```

And as far as the excluded tags, this only validates classes so they should not be an issue.

Reply:
I get it. I wanted to create separate plugins just in case if we need some more customizations on this. The one which you have mentioned is definately one way to go.

Reply:
This rule is not sufficient to handle both Utlity and Component type SUIT pattern. Therefor a plugin was needed to incorporate both of them.
I have made the Component type SUIT to always start with .spectrum. Please let me know whether we can initialize a classname with Component Name only also.
@pfulton